unison-src/transcripts/add-run.output.md

```diff
@@ -0,0 +1,292 @@
+```ucm
+.> add.run foo.bar.baz
+
+  ⍟ I've added these definitions:
+  
+    foo.bar.baz : Boolean
+
+```
+```ucm
+.> view foo.bar.baz
+
+  foo.bar.baz : Boolean
+  foo.bar.baz = true
+
+```
+## It resolves references within the unison file
+
+```unison
+z b = b Nat.+ 12
+y a b = a Nat.+ b Nat.+ z 10
+
+
+
+
+main : '{IO, Exception} (Nat -> Nat -> Nat)
+main _ = y
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      main : '{IO, Exception} (Nat -> Nat -> Nat)
+      y    : Nat -> Nat -> Nat
+      z    : Nat -> Nat
+
+```
+```ucm
+.> run main
+
+  a b -> a Nat.+ b Nat.+ z 10
+
+.> add.run result
+
+  ⍟ I've added these definitions:
+  
+    result : Nat -> Nat -> Nat
+    z      : Nat -> Nat
+
+```
+## It resolves references within the codebase
+
+```unison
+inc : Nat -> Nat
+inc x = x + 1
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      inc : Nat -> Nat
+
+```
+```ucm
+.> add inc
+
+  ⍟ I've added these definitions:
+  
+    inc : Nat -> Nat
+
+```
+```unison
+main : '(Nat -> Nat)
+main _ x = inc x
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      main : '(Nat -> Nat)
+
+```
+```ucm
+.> run main
+
+  inc
+
+.> add.run natfoo
+
+  ⍟ I've added these definitions:
+  
+    natfoo : Nat -> Nat
+
+.> view natfoo
+
+  natfoo : Nat -> Nat
+  natfoo = inc
+
+```
+## It captures scratch file dependencies at run time
+
+```unison
+x = 1
+y = x + x
+main = 'y
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      main : 'Nat
+      x    : Nat
+      y    : Nat
+
+```
+```ucm
+.> run main
+
+  2
+
+```
+```unison
+x = 50
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      x : Nat
+
+```
+this saves 2 to xres, rather than 100
+```ucm
+.> add.run xres
+
+  ⍟ I've added these definitions:
+  
+    xres : Nat
+
+.> view xres
+
+  xres : Nat
+  xres = 2
+
+```
+## It fails with a message if add cannot complete cleanly
+
+```unison
+main = '5
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      main : 'Nat
+
+```
+```ucm
+.> run main
+
+  5
+
+.> add.run xres
```

Comment:
Hrmm, I wonder if this error message might want a tweak; as a user I'd be confused about how I can `update` 'xres' since there's no `update.run` command 🤔 

Generally I think we want to provide a path forwards for folks, so I guess we could suggest deleting the existing `xres`?